pkg/pillar/cmd/volumemgr/blob.go

```diff
@@ -219,20 +420,33 @@ func getBlobChildren(blob *types.BlobStatus) []*types.BlobStatus {
 			return nil
 		}
 		log.Infof("getBlobChildren(%s): adding manifest %s", blob.Sha256, manifest.Digest.Hex)
-		return []*types.BlobStatus{
-			{
-				DatastoreID: blob.DatastoreID,
-				RelativeURL: replaceSha(blob.RelativeURL, manifest.Digest),
-				Sha256:      strings.ToLower(manifest.Digest.Hex),
-				Size:        uint64(manifest.Size),
-				State:       types.INITIAL,
-				BlobType:    types.BlobManifest,
-			},
+		childHash := strings.ToLower(manifest.Digest.Hex)
+		//Check if childBlob already exists
+		existingChild := lookupOrCreateBlobStatus(ctx, sv, childHash)
```

Comment:
This is a good catch. It really shouldn't create a new blob status, so much as check if one exists.